modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -164,23 +167,51 @@ function ActionItem({
 
 	const onActionDropdownItemClick = context.onActionDropdownItemClick;
 
-	return (
-		<ClayDropDown.Item
-			href={link ? url : null}
-			onClick={(event) => {
-				if (onActionDropdownItemClick) {
-					onActionDropdownItemClick({
-						action,
-						event,
-						itemData,
-					});
-				}
+	const handleClick = (event) => {
+		if (onActionDropdownItemClick) {
+			onActionDropdownItemClick({
+				action,
+				event,
+				itemData,
+			});
+		}
 
-				if (!link) {
-					handleClickOnLink(event);
-				}
-			}}
-		>
+		if (!link) {
+			handleClickOnLink(event);
+		}
+	};
+
+	const getQuickActionItem = () => {
+		const commonProps = {
+			'aria-label': icon,
+			'className': 'component-action quick-action-item',
+			'title': label,
+		};
+
+		if (link) {
+			return (
+				<ClayLink {...commonProps} href={url} onClick={handleClick}>
+					<ClayIcon symbol={icon} />
+				</ClayLink>
+			);
+		}
+		else {
+			return (
+				<ClayButtonWithIcon
+					{...commonProps}
+					displayType="unstyled"
+					monospaced={false}
+					onClick={handleClick}
+					symbol={icon}
+				/>
+			);
+		}
+	};
+
+	return quickActionItem ? (
+		getQuickActionItem()
+	) : (
```

Comment:
Should a quick action be a variant of `ActionItem`? I feel like these two could be completely separate components, maybe a `QuickAction`, or just `LinkOrButton`. Is there any benefit in keeping them merged?

Reply:
Between `quickActionItem` and `actionItem` the only thing changes is the markup, because we do not want to render always a dropdown item component, although it does the job...
They both share the logic for the event handler, that's why I keep them together